File: monte_pi/src/monte_pi_serial.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main (int argc, char *argv[])
{
  unsigned long long int i, num_points, count;
  double x, y, pi;
  clock_t t_start, t_end;
  
  num_points=300000000;
  count=0;
  t_start = clock();
  srand((unsigned) t_start);
  for(i=0;i<num_points;i++)
    {
      x = (double)rand()/RAND_MAX;
      y = (double)rand()/RAND_MAX;
      if (x*x + y*y <=1) count++;
    }
  
  pi = 4.0*count/num_points;
  
  t_end = clock();
  printf("Value of pi = %g\n",pi);
  printf("Expended wall clock time = %.5f\n", (double)(t_end - t_start)/CLOCKS_PER_SEC);

  return EXIT_SUCCESS;
    
}

File: monte_pi/src/monte_pi_serial.f90
PROGRAM monte_pi_serial

  IMPLICIT NONE
  
  INTEGER(kind=4) :: i, count
  REAL(kind=8) :: x, y, pi
  REAL(kind=8) :: t_start, t_end
  INTEGER(kind=8), PARAMETER :: num_points = 300000000

  count=0

  CALL cpu_time(t_start)
  CALL random_seed()
  DO i=0,num_points
     CALL random_number(x)
     CALL random_number(y)
     IF (x*x + y*y <= 1)  count = count + 1
  END DO

  pi = 4.0*count/num_points
  
  CALL cpu_time(t_end)

  WRITE(*,*) "Value of pi = ",pi
  WRITE(*,*) "Expended wall clock time = ",t_end-t_start
  
END PROGRAM monte_pi_serial
